### PR TITLE
Improve MessageListFragment and MessageListActivity

### PR DESCRIPTION
**Pull request by skydoves**
### 🎯 Goal
Improve `MessageListFragment` and `MessageListActivity`.

### 🛠 Implementation details
- Improved KDocs for `MessageListFragment`.
- Added `ThreadSafetyModes` for lazy properties to specify an initialization thread.
- Added `JvmStatic` and `JvmOverloads` on `createIntent` and `newInstance` methods for supporting Java side.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### 🎉 GIF
gif

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/MessageListFragment.kt [hunk 1]
```diff
@@ -40,18 +40,39 @@ import io.getstream.chat.android.ui.message.list.viewmodel.factory.MessageListVi
...
         requireNotNull(requireArguments().getString(ARG_CHANNEL_ID)) { "Channel cid must not be null" }
     }
-    protected val themeResId: Int by lazy { requireArguments().getInt(ARG_THEME_RES_ID) }
-    protected val messageId: String? by lazy { requireArguments().getString(ARG_MESSAGE_ID) }
-    protected val showHeader: Boolean by lazy { requireArguments().getBoolean(ARG_SHOW_HEADER, false) }
 
-    protected val factory: MessageListViewModelFactory by lazy { MessageListViewModelFactory(cid, messageId) }
-    protected val messageListViewModel: MessageListViewModel by viewModels { factory }
+    /** A custom theme resource for the screen. */
+    protected val themeResId: Int by lazy(lazyThreadMode) { requireArguments().getInt(ARG_THEME_RES_ID) }
```
Comment (samiuelson) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/MessageListFragment.kt:51]: I'm curious, does removing synchronization increase performance here?

My personal taste is that it's not necessary to extract `LazyThreadSafetyMode.NONE` enum to a separate `val`. Maybe code would be more readable if enum was applied directly to lazy, like `lazy(LazyThreadSafetyMode.NONE)`?. 

Another option would be to create new function like `fun unsafeLazy(initializer) = lazy(LazyThreadSafetyMode.NONE, initializer)`?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/MessageListFragment.kt [hunk 2]
```diff
@@ -140,23 +161,21 @@ public open class MessageListFragment : Fragment() {
...
+                    messageInputViewModel.setActiveThread(it.parentMessage)
+                }
+                is MessageListViewModel.Mode.Normal -> {
+                    messageListHeaderViewModel.resetThread()
+                    messageInputViewModel.resetThread()
                 }
             }
-            binding.messageListView.setMessageEditHandler(::postMessageToEdit)
         }
+        binding.messageListView.setMessageEditHandler(messageInputViewModel::postMessageToEdit)
```
Comment (samiuelson) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/MessageListFragment.kt:178]: Good catch!

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/MessageListFragment.kt (hunk 1 above)
Comment (skydoves) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/MessageListFragment.kt:51]: Exactly! This does removing synchronization works internally that increase performance when initializing them.
I agree the `LazyThreadSafetyMode` does not need to be extracted separately. 👍

Comment (skydoves) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/MessageListFragment.kt]: Applied in 7a48d9131

**Review by samiuelson — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**